Mathlib/GroupTheory/IndexNormal.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+import Mathlib.Data.Fintype.Perm
+import Mathlib.Data.Nat.Prime.Factorial
+import Mathlib.GroupTheory.Index
+
+/-! # Subgroups of small index are normal
+
+* `Subgroup.normal_of_index_eq_smallest_prime_factor`: in a finite group `G`,
+a subgroup of index equal to the smallest prime factor of `Nat.card G` is normal.
+
+* `Subgroup.normal_of_index_two`: in a group `G`, a subgroup of index 2 is normal
+(This does not require `G` to be finite.)
+
+-/
+
+namespace Nat
+
+theorem card_perm {α : Type*} [Finite α] : Nat.card (Equiv.Perm α) = (Nat.card α)! := by
+  classical
+  have := Fintype.ofFinite α
+  rw [Nat.card_eq_fintype_card, Nat.card_eq_fintype_card, Fintype.card_perm]
```

Comment:
This should probably be moved to a more appropriate file.

Reply:
sure, but i couldn't find any that doesn't require new imports...
do you have any suggestion?
creating a GroupTheory/Perm/Finite ?

Reply:
If there's no better place, then that seems like a reasonable option.